relay-server/src/actors/envelopes.rs

```diff
@@ -379,6 +379,9 @@ struct ProcessEnvelopeState {
     /// The state of the project that this envelope belongs to.
     project_state: Arc<ProjectState>,
 
+    /// The state of the project that initiated the current trace.
+    sampling_project_state: Option<Arc<ProjectState>>,
```

Comment:
Just personal preference, so feel free to ignore. If the variable is storing a `ProjectState`, I don't understand what the `sampling_` prefix is for.
```suggestion
    project_state: Option<Arc<ProjectState>>,
```

Reply:
See definition on line 380. I added a comment now to make the purpose of this extra field more clear.